### ISSUE TITLE
Wire up & test new VFS syscalls

### ISSUE THREAD

**Issue by mpe** [linuxppc/linux#167]
In linux-next there's multiple new syscalls being developed.

These may not all get merged, but it looks likely and we would ideally test them before they hit mainline.

There's at least:
 - `fsopen()`
 - `fsmount()`
 - `fspick()`

Then there's RFCs for:
 - `fsinfo()`
 - `move_mount()`

**Comment by leitao** [comment]
I started to take a look on this issue. I was able to wire up the syscalls and start testing.

[https://github.com/leitao/linux/commit/c06f61587bbc9fcda2597e4fc68a6244f720f807](https://github.com/leitao/linux/commit/c06f61587bbc9fcda2597e4fc68a6244f720f807)

Regarding the tests, I talked to David and he provided me userspace tests that I am adapting to get it running on Powerpc. 

I am keeping the userspace tests at [https://github.com/leitao/mcontext](https://github.com/leitao/mcontext)

**Comment by leitao** [comment]
Michael, I was able to wire up and test all the 5 new syscalls above as a new one that is being integrated called 'fsconfig'[1].

I created two patches, one to wire up the initial 5 that are on linux-next already and another commit for 'fsconfig'. They are:

[https://github.com/leitao/linux/commit/ae2fce8aa8355bcb4e78826b8fbf13cb0ff0ef7b ](https://github.com/leitao/linux/commit/ae2fce8aa8355bcb4e78826b8fbf13cb0ff0ef7b )
and
[https://github.com/leitao/linux/commit/ca3e9acf815264d5ccce3a3c18ba978550d42ab4](https://github.com/leitao/linux/commit/ca3e9acf815264d5ccce3a3c18ba978550d42ab4)

Do you want me to send them to the linux-ppc mailing list for public review?


[1][ https://git.kernel.org/pub/scm/linux/kernel/git/dhowells/linux-fs.git/commit/?h=mount-context&id=5e3289647144d4a2cb9f7a3ecc46c77335525d7a]( https://git.kernel.org/pub/scm/linux/kernel/git/dhowells/linux-fs.git/commit/?h=mount-context&id=5e3289647144d4a2cb9f7a3ecc46c77335525d7a)

**Comment by mpe** [comment]
Yeah those look good to me, send them to the list & cc LKML & dhowells. :+1:

**Comment by leitao** [comment]
Thanks Michael. Patch sent as you requested.

[https://lkml.org/lkml/2018/7/27/657](https://lkml.org/lkml/2018/7/27/657)
